lib/Diagnostics.php

```diff
@@ -0,0 +1,112 @@
+<?php
+/**
+ * @author Piotr Mrowczynski <piotr@owncloud.com>
+ *
+ * @copyright Copyright (c) 2017, ownCloud GmbH
+ * @license AGPL-3.0
+ *
+ * This code is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License, version 3,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License, version 3,
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>
+ *
+ */
+
+namespace OCA\Diagnostics;
+
+use OCP\IConfig;
+
+/**
+ * This class provides access to system and application configuration, is also responsible for 
+ * storing the data in the required destination as specified by the administrator
+ *
+ * @package OCA\Diagnostics
+ */
+class Diagnostics {
+	/** Nothing (collecting but not used) */
+	const LOG_NOTHING = '0';
+	/** Summary (one report per request) */
+	const LOG_SUMMARY = '1';
+	/** Everything (single queries with their parameters) */
+	const LOG_ALL = '2'; 
+
+	/** @var \OCP\IConfig */
+	private $config;
+	
+	/**
+	 * @param \OCP\IConfig $config
+	 */
+	public function __construct(IConfig $config) {
+		$this->config = $config;
+	}
+
+	/**
+	 * @return bool
+	 */
+	public function isDebugEnabled() {
+		return $this->config->getSystemValue('debug', false);
+	}
+
+	/**
+	 * @param bool $enable
+	 */
+	public function setDebug($enable) {
+		$this->config->setSystemValue('debug', $enable);
+	}
+	
+	/**
+	 * @return string
+	 */
+	public function getDiagnosticLogLevel() {
+		return $this->config->getAppValue('diagnostics', 'diagnosticLogLevel', Diagnostics::LOG_NOTHING);
+	}
+
+	/**
+	 * @param string $logLevel
+	 */
+	public function setDiagnosticLogLevel($logLevel) {
+		$this->config->setAppValue('diagnostics', 'diagnosticLogLevel', $logLevel);
+	}
+
+	/**
+	 * @param string $appName
+	 * @param string $sqlStatement
+	 * @param array $sqlParams
+	 * @param float $sqlQueryDuration
+	 */
+	public function recordQuery($appName, $sqlStatement, $sqlParams, $sqlQueryDuration) {
+		// For now, write to normal log file directly, and force INFO level
+		$message = 'DB CALL : '.str_replace("\"", "", str_replace("\t", "", str_replace("\n", " ", $sqlStatement)))
+			.' WITH : '.str_replace("\n", " ", var_export($sqlParams, true))
+			.' TOOK : '.strval($sqlQueryDuration).' MSEC';
+		\OC\Log\Owncloud::write($appName, $message,\OCP\Util::INFO);
```

Comment:
Problem is, I want to force log even when someone set logging level low. Idealy, I want to have custom log file maybe? Thus, is someone set diagnostics to log sth, it will be logged regardless of other settings.